File: SkaiLab.Invoice/ClientApp/src/app/models/ProfitAndLostDetail.ts
export class ProfitAndLostDetailItem {
    date: Date;
    transactionType: string;
    number: string;
    clientOrVendorName: string;
    description: string;
    amount: number;
    parentId:number;
    constructor() {
        this.date = new Date();
        this.transactionType = "";
        this.number = "";
        this.clientOrVendorName = "";
        this.description = "";
        this.amount = 0;
        this.parentId=0;
    }
}
export class ProfitAndLostDetailParent{
    name:string;
    total:number;
    isExpand:boolean;
    profitAndLostDetailItems:ProfitAndLostDetailItem[];
    constructor(){
        this.name="";
        this.total=0;
        this.isExpand=true;
        this.profitAndLostDetailItems=[];
    }
}
export class ProfitAndLostDetailTotal{
    name:string;
    total:number;
    constructor(){
        this.name="";
        this.total=0;
    }
}
export class ProfitAndLostDetail{
    profitAndLostDetailTotal:ProfitAndLostDetailTotal;
    profitAndLostDetailParents:ProfitAndLostDetailParent[];
    constructor(){
        this.profitAndLostDetailTotal=new ProfitAndLostDetailTotal();
        this.profitAndLostDetailParents=[];
    }
}

File: SkaiLab.Invoice.Models/Report/ProfitAndLostDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkaiLab.Invoice.Models.Report
{
    public class ProfitAndLostDetail
    {
        public ProfitAndLostDetailTotal ProfitAndLostDetailTotal { get; set; }
        public List<ProfitAndLostDetailParent> ProfitAndLostDetailParents { get; set; }

    }
    public class ProfitAndLostDetailTotal
    {
        public string Name { get; set; }
        public decimal Total { get; set; }
    }
    public class ProfitAndLostDetailParent
    {
        public string Name { get; set; }
        public decimal Total { get; set; }
        public bool IsExpand { get; set; }
        public List<ProfitAndLostDetailItem> ProfitAndLostDetailItems { get; set; }
    }
    public class ProfitAndLostDetailItem
    {
        public DateTime Date { get; set; }
        public string TransactionType { get; set; }
        public string Number { get; set; }
        public string ClientOrVendorName { get; set; }
        public string Description { get; set; }
        public decimal Amount { get; set; }
        public long ParentId { get; set; }
    }
}
